R/PipeOpDateFeatures.R

```diff
@@ -0,0 +1,209 @@
+#' * `hour` :: `logical(1)`\cr
+#'   Whether the hour should be extracted as a feature. Default TRUE.
+#' * `minute` :: `logical(1)`\cr
+#'   Whether the minute should be extracted as a feature. Default TRUE.
+#' * `second` :: `logical(1)`\cr
+#'   Whether the second should be extracted as a feature. Default TRUE.
+#' * `is_day` :: `logical(1)`\cr
+#'   Whether a feature should be extracted indicating whether it is day time (06:00 am - 08:00pm).
+#'   Default TRUE.
+#'
+#' @section Internals:
+#' If `cyclic = TRUE`, cyclic features are computed for the features `"month"`, `"week_of_year"`,
+#' `"day_of_year"`, `"day_of_month"`, `"day_of_week"`, `"hour"`, `"minute"` and `"second"`. This
+#' means that each feature, `x`, is replaced by two features, namely the sinus and cosinus
+#' transformation of `2 * pi * x / max(x)`. This is useful to respect the cyclical nature of
+#' features such as seconds, i.e., second 21 and second 22 are one second apart, but so are second
+#' 60 and second 01. The transformation always assumes that `min(x) = 0`, therefore prior shifting
+#' the values internally by minus one may occurr if necessary.
+#'
+#' @section Methods:
+#' Only methods inherited from [`PipeOpTaskPreprocSimple`]/[`PipeOp`].
+#'
+#' @examples
+#' library("mlr3")
+#' dat = iris
+#' set.seed(1)
+#' dat$date = sample(seq(from = as.POSIXct("2020-01-29"), to = as.POSIXct("2020-02-28"), by = "hour"),
+#'   size = 150L)
+#' task = TaskClassif$new("iris_date", backend = dat, target = "Species")
+#' pop = po("datefeatures", param_vals = list(date_variable = "date", cyclic = FALSE, minute = FALSE,
+#'   second = FALSE))
+#' pop$train(list(task))
+#' pop$state
+#' @family PipeOps
+#' @include PipeOpTaskPreproc.R
+#' @export
+PipeOpDateFeatures = R6Class("PipeOpDateFeatures",
+  inherit = PipeOpTaskPreprocSimple,
+  public = list(
+    initialize = function(id = "datefeatures", param_vals = list()) {
+      ps = ParamSet$new(params = list(
+        ParamUty$new("date_variable", custom_check = check_character, tags = "required"),
+        ParamLgl$new("cyclic", default = FALSE),
+        ParamLgl$new("year", default = TRUE, tags = "date_feature"),
+        ParamLgl$new("month", default = TRUE, tags = "date_feature"),
+        ParamLgl$new("week_of_year", default = TRUE, tags = "date_feature"),
+        ParamLgl$new("day_of_year", default = TRUE, tags = "date_feature"),
+        ParamLgl$new("day_of_month", default = TRUE, tags = "date_feature"),
+        ParamLgl$new("day_of_week", default = TRUE, tags = "date_feature"),
+        ParamLgl$new("hour", default = TRUE, tags = "date_feature"),
+        ParamLgl$new("minute", default = TRUE, tags = "date_feature"),
+        ParamLgl$new("second", default = TRUE, tags = "date_feature"),
+        ParamLgl$new("is_day", default = TRUE, tags = "date_feature")
+      ))
+      ps$values$date_variable = "date" # needed for mlr_pipeops overview
```

Comment:
Do we have a convention here? Should this be NULL or something?

Reply:
Also I would just initialize all logical Params above to `TRUE`here.
i.e.
`ps$values = list(xyz = TRUE, xyz2 = TRUE, ...)`

Reply:
in the updated version I initialize to `character(0L)` but I guess this is still not optimal?